cmsplugin_feed_ai/feed.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+from dateutil.parser import parse as parse_datetime
+from django.conf import settings
+from django.core.cache import cache
+from django.core.exceptions import ImproperlyConfigured
+from facebook import GraphAPI
+from twython import Twython
+from warnings import warn
+
+
+DEFAULT_CACHE_TIME = 3600  # 1 hour
+
+
+class FeedDisplayItem(object):
+    created_at = None
+    html_content = ""
+    content_link = ""
+    profile_name = ""
+    source = ""
+    source_id = ""
+    text_content = ""
+
+    def __init__(self, feed_obj, profile_name, source):
+        """
+        Initializer display item from raw feed object provided by
+        service specific API tool.
+        """
+        self.profile_name = profile_name
+        self.source = source
+        if source == "twitter":
+            self._init_from_twitter_object(feed_obj)
```

Comment:
Mmm, these should be subclasses so one class won't need to know about different providers.

```python
class FeedDisplayItem(object):
   ...

class TwitterFeedDisplayItem(FeedDisplayItem):
   source = "twitter"
   # ctor has logic for populating from twitter object

class FacebookFeedDisplayItem(...

```